sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/v2Commands.scala

```diff
@@ -404,8 +404,17 @@ case class MergeIntoTable(
     mergeCondition: Expression,
     matchedActions: Seq[MergeAction],
     notMatchedActions: Seq[MergeAction]) extends BinaryCommand with SupportsSubquery {
+
+  override lazy val resolved: Boolean = {
```

Comment:
this fixes another bug where a merge plan with star and no other expressions with unresolved attributes (e.g. all non-optional predicates are `literal(true)`), then the resolution will be skipped and the star won't be expanded. added test for that.

Reply:
How about we simply mark `InsertStarAction` and `UpdateStarAction` as unresolved as they must be expanded during analysis? Similar to `Star`

Reply:
oh yeah. thats much better!